src/sentry/api/endpoints/event_file_committers.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+import six
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import (
+    Release, ReleaseCommit, Commit, CommitFileChange, Event
+)
+from sentry.api.serializers.models.release import get_users_for_commits
+
+from django.db.models import Q
+
+from itertools import izip
+from collections import defaultdict
+
+
+def tokenize_path(path):
+    # TODO(maxbittker) tokenize in a smarter crossplatform way.
+    return reversed(path.split('/'))
+
+
+class EventFileCommittersEndpoint(ProjectEndpoint):
+
+    def _get_frame_paths(self, event):
+        data = event.data
+        try:
+            frames = data['sentry.interfaces.Stacktrace']['frames']
+        except KeyError:
+            try:
+                frames = data['sentry.interfaces.Exception']['values'][0]['stacktrace']['frames']
+            except KeyError:
+                return []  # can't find stacktrace information
+
+        return frames
+
+    def _score_path_match_length(self, path_a, path_b):
+        score = 0
+        for a, b in izip(tokenize_path(path_a), tokenize_path(path_b)):
+            if a != b:
+                break
+            score += 1
+        return score
+
+    def _get_commits(self, project_id, version):
+        try:
+            commits = Commit.objects.filter(
+                releasecommit=ReleaseCommit.objects.filter(
+                    release=Release.objects.get(
+                        projects=project_id,
+                        version=version,
+                    ),
+                )
+            )
+        except Release.DoesNotExist:
+            return None
+        return list(commits)
+
+    def _get_commit_file_changes(self, commits, path_name_set):
```

Comment:
This would be much more readable:

```python
queryset = CommitFileChange.objects.filter(
    commit__in=commits
).filter(
    reduce(operators.or_, (
        Q(filename__endswith=next(tokenize_path(path)))
        for path in path_name_set
    ))
)

return list(queryset)
```

Would need to for sure confirm if the behavior is the same. The way you're doing it is much more convoluted.